src/components/AnnotationProps.tsx

```diff
@@ -42,6 +42,37 @@ export default interface AnnotationProps {
    */
   onDeselect?: () => void;
 
+  /**
+   * Event fired with the annotation is drag start.
+   */
+  onDragStart?: () => void;
+
+  /**
+   * Event fired with the annotation is drag end.
+   */
+  onDragEnd?: () => void;
+
+  /**
+   * A Boolean value that determines whether the map animates the annotation.
+   */
+  animates?: boolean;
+
+  /**
+   * A CSS animation that runs when the annotation appears on the map.
+   * @see {@link https://developer.apple.com/documentation/mapkitjs/mapkit/annotation/2973818-appearanceanimation}
+   */
+  appearanceAnimation?: string;
```

Comment:
The changes related to animations are unrelated to the rest of the PR. In this situation, you should submit two separate pull requests, so that the changes are easier to review. (You don’t need to split this particular PR, but it would be better in the future.)

Reply:
Fully agree with this one, though it was a low hanging fruit to implement it with the other changes and moving it to a separate PR potentially generates future merge conflicts when having too many PRs at the same time. Thank you so much!